packages/aws-cdk-lib/pipelines/lib/helpers-internal/pipeline-graph.ts

```diff
@@ -134,7 +134,12 @@ export class PipelineGraph {
     const stackGraphs = new Map<StackDeployment, AGraph>();
 
     for (const stack of stage.stacks) {
-      const stackGraph: AGraph = Graph.of(this.simpleStackName(stack.stackName, stage.stageName), { type: 'stack-group', stack });
+      const stackGraphName = findUniqueName(retGraph, [
+        this.simpleStackName(stack.stackName, stage.stageName),
+        ...stack.account ? [stack.account] : [],
+        ...stack.region ? [stack.region] : [],
+      ]);
```

Comment:
nit picking: Just thinking of an edge case when 4 stack names are the same, then this won't fix it right?

Reply:
But a different stack couldn't have the same name AND be deployed to both the same region AND account. Then it would just be the same stack.